### ISSUE TITLE
Resuming incremental search replaces old search with word at current location

### ISSUE THREAD

**Issue by m27315** [whitphx/vscode-emacs-mcx#107]
## Steps to Reproduce
1. Search for word (C-s).
2. Exit incremental search through movement (e.g., C-a, jump to front of line).
3. Assuming a different word is now under the cursor, resuming the incremental search (pressing C-s again) will instead search for the current word instead of resuming previous search.

## Expected Behavior
Emacs would resume the previous incremental search, instead of starting a new search with the word under the cursor.

## Versions
```
Awesome Emacs Keymap: 0.11.2
VSCode Version: 1.33.1
Commit: 51b0b28134d51361cf996d2f0a1c698247aeabd8
Date: 2019-04-11T08:20:22.771Z
Electron: 3.1.6
Chrome: 66.0.3359.181
Node.js: 10.2.0
V8: 6.6.346.32
OS: Linux x64 4.19.27-gentoo-r1
```

**Comment by joeshaw** [comment]
Setting `editor.find.seedSearchStringFromSelection` to false will stop the find widget from doing this.  Perhaps the extension should set this setting.

**Comment by tuttieee** [comment]
Overriding configuration by extension may not be allowed now.
Related issue: https://github.com/microsoft/vscode/issues/26707#issuecomment-461538461

For now, please set `editor.find.seedSearchStringFromSelection` config as @joeshaw wrote. Thanks!

**Comment by TTimo** [comment]
I love the extension, but this particular problem with C-s has been driving me nuts. Thanks for the workaround!

**Comment by tuttieee** [comment]
memo: https://github.com/microsoft/vscode/issues/1396

**Comment by laktech** [comment]
https://code.visualstudio.com/docs/getstarted/settings#_default-settings

link to the config docs as it's not a boolean type as the comments above suggest. the config accepts "never", "always", "selection"

**Comment by whitphx** [comment]
Thank you.
It seems to have been changed at v1.59: https://github.com/microsoft/vscode-docs/commit/a27b0efea3c05b4e06548dba1ad6b58979f1a79a#diff-d432cc9e616a920ea35cfeddb72b780f663270fd96f6871ec99dc8bda5b8cfb5R350

I will update the README of this extension.

**Comment by whitphx** [comment]
Yet, [this page](seedSearchStringFromSelection) still says
> This feature can be turned off by setting editor.find.seedSearchStringFromSelection to false.

So it seems to accept both `false` and `"never"`.

**Comment by whitphx** [comment]
Updating the VSCode settings seems to be possible:
* https://stackoverflow.com/questions/68526627/vscode-extension-how-do-i-edit-workspace-global-config
* https://stackoverflow.com/questions/54356069/vs-code-extension-to-update-workspace-settings

-> Try to set `editor.find.seedSearchStringFromSelection` programatically.

**Comment by whitphx** [comment]
From 0.54.0, this extension overrides the default value of `editor.find.seedSearchStringFromSelection` to `"never"` so the workaround introduced above will no longer necessary.
It still allows configuring the setting by users, so people who like VSCode's default behavior like #1278 can use it while they need to edit the settings by hand.

**Comment by vzaliva** [comment]
Setting it to "never" still does not solve this issue. It is no longer initialised with selection, but after moving the cursor and re-entering the search, it loses the previous search term.

**Comment by cjauvin** [comment]
Just an anecdote about this: after having been sufficiently annoyed by this behavior (without knowing anything about it here), I decided to spend a few hours trying to implement a fix by myself. I thought I had made good progress (knowing nothing about VSC extensions, ChatGPT and Cursor have been quite helpful), but then realized that there's no API (apparently?) to retrieve the text from the find widget, which was necessary for what I had in mind. I thought I was blocked, but then I came here, and discovered about the `editor.find.seedSearchStringFromSelection` option, which I had no idea about (I was seeing it in the code, but ignoring it, because it didn't feel relevant), and which solves my problem!
